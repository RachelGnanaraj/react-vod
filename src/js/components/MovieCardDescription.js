import React from 'react';

export default class MovieCardDescription extends React.Component {
  getDescription = (p = this.props) => {
    if (p.expanded || p.description.length < 100) return p.description
    else {
      if (p.description.length > 100) {
        return p.description.slice(0, 100) + '...'
      }
    }
  }

  render() {
    return (
        <p className="card-text movie-description">{this.getDescription()}</p>
    )
  }
}
