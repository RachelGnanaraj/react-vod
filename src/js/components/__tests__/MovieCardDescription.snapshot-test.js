import React from 'react';
import MovieCardDescription from '../MovieCardDescription';
import renderer from 'react-test-renderer';

const longText = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'

test('MovieCardDescription render full text when hovered', () => {
    const component = renderer.create(
        <MovieCardDescription description={longText} expanded={true}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('MovieCardDescription slice text when not hovered', () => {
    const component = renderer.create(
        <MovieCardDescription description={longText} expanded={false}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});