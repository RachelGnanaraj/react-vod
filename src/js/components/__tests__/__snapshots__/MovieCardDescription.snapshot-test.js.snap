// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardDescription render full text when hovered 1`] = `
<p
  className="card-text movie-description"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</p>
`;

exports[`MovieCardDescription slice text when not hovered 1`] = `
<p
  className="card-text movie-description"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore ...
</p>
`;
